src/libtiled/maprenderer.cpp

```diff
@@ -48,6 +48,21 @@
 
 using namespace Tiled;
 
+// from https://github.com/anak10thn/qt5-tutorials/blob/master/imagetint/imagetint.cpp
+// Tint an image with the specified color and return it as a new image
+QImage tinted(const QImage &image, const QColor &color, QPainter::CompositionMode mode = QPainter::CompositionMode_Screen)
+{
+    QImage resultImage(image.size(), QImage::Format_ARGB32_Premultiplied);
+    QPainter painter(&resultImage);
+    painter.drawImage(0, 0, image);
+    painter.setCompositionMode(mode);
```

Comment:
Since I don't think we're planning to support other composition modes, please just put `QPainter::CompositionMode_Multiply` here.